Cargo.toml

```diff
@@ -34,11 +34,13 @@ tempfile = "3.1.0"
 [features]
 unstable = []
 default = ["ssh", "https", "ssh_key_from_memory"]
-ssh = ["libgit2-sys/ssh"]
+ssh = ["ssh-libssh2"]
+ssh-libssh2 = ["libgit2-sys/ssh-libssh2"]
+ssh-openssh = ["libgit2-sys/ssh-openssh"]
```

Comment:
If both features are set, I think libssh2 will take precedence, because we are just adding `#define` instructions, and the `#ifdef` in the libgit2 codebase appears to always check for libssh2 first. ([example](https://github.com/libgit2/libgit2/blob/6c5520f334e5652d5f0476c00a3188d1d97754e7/src/libgit2/transports/ssh.c#L18)).

If we do (3), my understanding is that we would need to add more environment variables similar to `LIBGIT2_NO_VENDOR` to allow explicit disabling of one of the features? (Correct me if I'm wrong.) This makes it seems like doing (1) + (2) would be better.